OSDOCS-14645: disconnected env is supported in SSCSI (#93255)
openshift/openshift-docs: modules/persistent-storage-csi-secrets-store-disconnect-environment.adoc

// Module included in the following assemblies:
//
// * storage/container_storage_interface/persistent-storage-csi-secrets-store.adoc
//

:_mod-docs-content-type: CONCEPT
[id="persistent-storage-csi-secrets-store-disconnect-environment_{context}"]
= Support for disconnected environments

> The {secrets-store-driver} is verified to work in disconnected clusters with the following secrets store providers:

Reviewer: Suggestion to change from passive to active voice: 
```suggestion
The following secrets store providers support the {secrets-store-driver} in disconnected clusters:
```

> The {secrets-store-driver} now supports disconnected environment. The {secrets-store-driver} is verified to work in the disconnected clusters with the following supported secrets providers:

Reviewer 2: The word `now` sounds lil misleading to me. It's like disconnected is only supported starting from 4.19 and wasn't supported in previous versions. I think our intention should be that we're claiming its supportability from 4.19+.

/cc @mytreya-rh 
```suggestion
The {secrets-store-driver} is verified to work in the disconnected clusters with the following supported secrets providers:
```

Author: Sentence is updated.

* link:https://aws.amazon.com/secrets-manager/[AWS Secrets Manager]
* link:https://developer.hashicorp.com/vault[HashiCorp Vault]
* link:https://cloud.google.com/security/products/secret-manager[Google Secrets Manager]
* link:https://azure.microsoft.com/en-us/products/key-vault/[Azure Key Vault]

> To enable communication between {secrets-store-driver} and the secrets manager, configure VPC-endpoints or equivalent connectivity to the corresponding secrets manager service. Additionally, configure VPC-endpoints or connectivity to the OpenID(OIDC) issuer and the Secure Token Service(STS). The exact configuration depends on the secrets provider, the authentication method and the types of disconnected cluster.

Reviewer 2: I see mixed usage of `secrets manager` and `secrets provider`.

Also `Additionally, configure VPC-endpoints` sounds redundant to the previous sentence.
